torchgeo/datasets/deepglobelandcover.py

```diff
@@ -0,0 +1,267 @@
+    semantic segmentation to detect areas of urban, agriculture, rangeland, forest,
+    water, barren, and unknown. It contains 1,146 satellite images of size
+    2448 x 2448 pixels in total, split into training/validation/test sets, the original
+    dataset can be downloaded from `Kaggle <https://www.kaggle.com/datasets/balraj98/
+    deepglobe-land-cover-classification-dataset>`__.
+    However, we only use the training dataset with 803 images since the original test
+    and valid dataset are not accompanied by labels. The dataset that we use with a
+    custom train/test split can be downloaded from `Kaggle <https://www.kaggle.com/
+    datasets/geoap96/deepglobe2018-landcover-segmentation-traindataset>`__ (created as a
+    part of Computer Vision by Deep Learning (CS4245) course offered at TU Delft).
+
+    Dataset format:
+
+    * images are RGB data
+    * masks are RGB image with with unique RGB values representing the class
+
+    Dataset classes:
+
+    0. Urban land
+    1. Agriculture land
+    2. Rangeland
+    3. Forest land
+    4. Water
+    5. Barren land
+    6. Unknown
+
+    File names for satellite images and the corresponding mask image are id_sat.jpg and
+    id_mask.png, where id is an integer assigned to every image.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/pdf/1805.06561.pdf
+
+    """
+
+    filename = "data.zip"
+    data_root = "data"
+    md5 = "f32684b0b2bf6f8d604cd359a399c061"
+    splits = ["train", "test"]
+    classes = [
+        "Urban land",
+        "Agriculture land",
+        "Rangeland",
+        "Forest land",
+        "Water",
+        "Barren land",
+        "Unknown",
+    ]
+    colormap = [
+        (0, 255, 255),
+        (255, 255, 0),
+        (255, 0, 255),
+        (0, 255, 0),
+        (0, 0, 255),
+        (255, 255, 255),
+        (0, 0, 0),
+    ]
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new DeepGlobeLandCover dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train" or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+        """
+        assert split in self.splits
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.checksum = checksum
+
+        self._verify()
+        if split == "train":
+            split_folder = "training_data"
+        else:
+            split_folder = "test_data"
+
+        self.files = []
+        for image in os.listdir(
```

Comment:
The order of files returned by `os.listdir` is non-deterministic. It might be wise to sort `self.files` by image key at the end to make things more reproducible.

Reply:
Added `sorted()` around `os.listdir()` instead